communication/mavlink_communication.hpp

```diff
@@ -104,139 +91,79 @@ class Mavlink_communication
     Mavlink_communication(Serial& serial, State& state, File& file_storage, const conf_t& config = default_config());
 
     /**
-     * \brief   Adding new message to the MAVLink scheduler
-     *
-     * \param   repeat_period           Repeat period (us)
-     * \param   run_mode                Run mode
-     * \param   timing_mode             Timing mode
-     * \param   priority                Priority
-     * \param   function                Function pointer to be called
-     * \param   module_structure        Argument to be passed to the function
-     * \param   task_id                 Unique task identifier
-     *
-     * \return  True if the message was correctly added, false otherwise
-     */
-    bool add_msg_send(uint32_t task_id,
-                      uint32_t repeat_period,
-                      send_msg_function_t function,
-                      handling_telemetry_module_struct_t module_structure,
-                      Scheduler_task::priority_t priority       = Scheduler_task::PRIORITY_NORMAL,
-                      Scheduler_task::timing_mode_t timing_mode = Scheduler_task::PERIODIC_RELATIVE,
-                      Scheduler_task::run_mode_t run_mode       = Scheduler_task::RUN_REGULAR);
-
-
-
-    /**
-     * \brief   Suspending sending of messages
-     *
-     * \param   delay                   Delay of suspension in microsecond
-     */
-    void suspend_downstream(uint32_t delay);
-
-
-    /**
      * \brief   Returns sysid of the underlying mavlink_stream
      *
      *
      * \return  sysid of the underlying mavlink_stream
      */
      uint32_t sysid();
 
+
     /*
-     * \brief   Returns scheduler
+     * \brief   Returns periodi telemetry
      */
-    Scheduler& scheduler();
+    Periodic_telemetry& telemetry();
```

Comment:
I believe not, you can do either `p = &com.telemetry()` of `p = com.p_telemetry()`